numpy/testing/_private/utils.py

```diff
@@ -706,6 +711,9 @@ def chk_same_position(x_id, y_id, hasval='nan'):
                                 % (hasval), verbose=verbose, header=header,
                                 names=('x', 'y'), precision=precision)
             raise AssertionError(msg)
+        # If there is a scalar, then here we know the array has the same
+        # flag as it everywhere, so we should return the scalar flag.
+        return x_id if not x_id.shape else y_id
```

Comment:
nit: I think this is clearer as `x_id.ndim == 0` rather than `not x_id.shape`

Reply:
Yes, done.